azure/dns-records/main.tf

```diff
@@ -0,0 +1,45 @@
+data "azurerm_dns_zone" "record" {
+  name                = var.dns_zone
+  resource_group_name = var.dns_rg
+}
```

Comment:
It might be a good idea to remove it, since it's a bit redundant. If you take a look into the code, this data block is basically fetching some data that you already have through the variables `var.dns_zone` and `var.dns_rg`

Reply:
you-re right, this was the first resource I-ve put and then while creating the rest didn-t noticed. Nice catch.